target/linux/ramips/dts/RB750Gr3.dts

```diff
@@ -67,63 +68,95 @@
 		};
 
 		usb {
-			gpio-export,name = "usb";
+			gpio-export,name = "usb_power_off";
 			gpio-export,output = <1>;
-			gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+
+		// Used to enable power-over-ethernet passthrough to eth4 (according to OEM source).
```

Comment:
Please add it to target/linux/ramips/base-files/etc/board.d/03_gpio_switches. This way it can be changed using the normal OpenWrt configuration utility uci (or via /etc/config/system)

Reply:
@mkresin, you do mean the usb power off switch only here, right? I'm having a hard time here, I am not familiar with the uci concept. If there is a document describing the uci for developers, please point me to it.

I gave it a first shot, added ucidef_add_gpio_switch to the 03_gpio_switches. Should the configuration parameter appear somewhere automatically?

Reply:
> you do mean the usb power off switch only here, right

I mean the "enable power-over-ethernet passthrough". The usb stuff need to be done via gpio-exports, as the kernel driver may fail to load if usb pwr is disabled.

>  If there is a document describing the uci for developers, please point me to it.

Your best bet is the OpenWrt homepage/wiki.

> I gave it a first shot, added ucidef_add_gpio_switch to the 03_gpio_switches. Should the configuration parameter appear somewhere automatically?

yes, they should be in /etc/config/*. I'm not entirely sure, but they should be in the system file.

Reply:
> > you do mean the usb power off switch only here, right
> I mean the "enable power-over-ethernet passthrough". The usb stuff need to be done via gpio-exports, as the kernel driver may fail to load if usb pwr is disabled.

Gotcha! That's exactly what I observed. 

> > If there is a document describing the uci for developers, please point me to it.
> Your best bet is the OpenWrt homepage/wiki.

Thanks. Went to look into.
 
> > I gave it a first shot, added ucidef_add_gpio_switch to the 03_gpio_switches. Should the configuration parameter appear somewhere automatically?
> 
> yes, they should be in /etc/config/*. I'm not entirely sure, but they should be in the system file.

Let's see...

Thanks for your patience with me.